backend/app/entity/shortlink.go

```diff
@@ -17,3 +17,44 @@ type ShortLink struct {
 	OpenGraphTags metatag.OpenGraph
 	TwitterTags   metatag.Twitter
 }
+
+// ShortLinkInput represents possible ShortLink attributes
+type ShortLinkInput struct {
+	LongLink    *string
+	CustomAlias *string
+	ExpireAt    *time.Time
+}
+
+// IsEmpty checks if ShortLinkInput has no attributes specified.
+func (s *ShortLinkInput) IsEmpty() bool {
+	return *s == ShortLinkInput{}
+}
+
+// GetLongLink fetches LongLink for ShortLinkInput, with empty string as the default value.
+func (s *ShortLinkInput) GetLongLink() string {
+	if s.LongLink == nil {
+		return ""
+	}
+	return *s.LongLink
+}
+
+// GetCustomAlias fetches CustomAlias for ShortLinkInput, with empty string as the default value.
+func (s *ShortLinkInput) GetCustomAlias() string {
+	if s.CustomAlias == nil {
+		return ""
+	}
+	return *s.CustomAlias
+}
+
+// GetShortLink creates a ShortLink entity instance using ShortLinkInput.
+func (s *ShortLinkInput) GetShortLink() *ShortLink {
+	if s.IsEmpty() {
+		return nil
+	}
+
+	return &ShortLink{
+		Alias:    s.GetCustomAlias(),
+		LongLink: s.GetLongLink(),
+		ExpireAt: s.ExpireAt,
+	}
+}
```

Comment:
I wonder whether this function is actually needed.

Reply:
The logic of getting `ShortLink` from `ShortLinkInput` was being done in two spots (in create and in update). Therefore, I think it makes sense to have this helper function create the `ShortLink` object and remove the duplicate logic.

Reply:
@Coteh I wonder whether we can eliminate the case of getting `ShortLink` from `ShortLinkInput`.

Reply:
I'm not sure because if I understand correctly the reason why we have `ShortLinkInput` is because alias or long link are optional parameters within the GraphQL API. We would have to update the signature for both `CreateShortLink` and `UpdateShortLink` methods to accept a `ShortLinkInput` instead of a `ShortLink` instance, if we want to eliminate getting `ShortLink` from `ShortLinkInput`.

Reply:
@Coteh I think updating the use cases is a great option. This delegated detailed decisions to the use case layer instead of forcing the API layer to do conversion. Once we converted `input` to non optional representation, the use case won't have the concept of "unset" field any more and won't be able to respond to those scenarios.

Reply:
I recommend updating the use case if possible.

Reply:
Leaving a comment for myself, @byliuyang clarified over Slack that the use case repository methods for `CreateShortLink` and `UpdateShortLink` should make the decision with short link input, rather than have API layer make the decision and pass it in which is what I have right now. I will resolve this comment when this has been changed.